functional-samples/tutorial.open-api-reference/sw-omnibox.js

```diff
@@ -0,0 +1,36 @@
+import { getAPIsuggestions } from './sw-api-list.js';
+
+console.log('sw-omnibox.js');
+
+// Save default API suggestions
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.storage.local.set({ apiSugg: ['tabs', 'storage', 'scripting'] });
+  }
+});
+
+const chromeURL = 'https://developer.chrome.com/docs/extensions/reference/';
+
+// Displays the suggestions after user starts typing
+chrome.omnibox.onInputChanged.addListener(async (input, suggest) => {
+  const suggestions = await getAPIsuggestions(input);
+  suggest(suggestions);
+});
+
+// Opens the reference page of the chosen API
+chrome.omnibox.onInputEntered.addListener(async (input) => {
+  await chrome.tabs.create({ url: chromeURL + input });
+  // Saves the latest keyword
+  updateHistory(input);
+});
+
+async function updateHistory(input) {
+  try {
+    const { apiSugg } = await chrome.storage.local.get('apiSugg');
+    apiSugg.unshift(input);
+    apiSugg.splice(4);
```

Comment:
nit: magic number - it'd be good to encode this in a constant named e.g. NUMBER_OF_PREVIOUS_SEARCHES